spec/requests/api/v1/invoice_items_spec.rb

```diff
@@ -0,0 +1,259 @@
+    expect(Time.zone.parse(found_invoice_item["updated_at"]).to_s).to eq(invoice_item.updated_at.to_s)
+  end
+
+  it "can return single record by unit_price" do
+    invoice_item = create(:invoice_item)
+
+    get "/api/v1/invoice_items/find?unit_price=#{invoice_item.unit_price}"
+
+    found_invoice_item = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(found_invoice_item["id"]).to eq(invoice_item.id)
+    expect(found_invoice_item["item_id"]).to eq(invoice_item.item_id)
+    expect(found_invoice_item["invoice_id"]).to eq(invoice_item.invoice_id)
+    expect(found_invoice_item["quantity"]).to eq(invoice_item.quantity)
+    expect(found_invoice_item["unit_price"]).to eq(invoice_item.unit_price)
+    expect(Time.zone.parse(found_invoice_item["created_at"]).to_s).to eq(invoice_item.created_at.to_s)
+    expect(Time.zone.parse(found_invoice_item["updated_at"]).to_s).to eq(invoice_item.updated_at.to_s)
+  end
+
+  it "can return single record by created_at" do
+    invoice_item = create(:invoice_item)
+
+    get "/api/v1/invoice_items/find?created_at=#{invoice_item.created_at}"
+
+    found_invoice_item = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(found_invoice_item["id"]).to eq(invoice_item.id)
+    expect(found_invoice_item["item_id"]).to eq(invoice_item.item_id)
+    expect(found_invoice_item["invoice_id"]).to eq(invoice_item.invoice_id)
+    expect(found_invoice_item["quantity"]).to eq(invoice_item.quantity)
+    expect(found_invoice_item["unit_price"]).to eq(invoice_item.unit_price)
+    expect(Time.zone.parse(found_invoice_item["created_at"]).to_s).to eq(invoice_item.created_at.to_s)
+    expect(Time.zone.parse(found_invoice_item["updated_at"]).to_s).to eq(invoice_item.updated_at.to_s)
+  end
+
+  it "can return single record by updated_at" do
+    invoice_item = create(:invoice_item)
+
+    get "/api/v1/invoice_items/find?updated_at=#{invoice_item.updated_at}"
+
+    found_invoice_item = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(found_invoice_item["id"]).to eq(invoice_item.id)
+    expect(found_invoice_item["item_id"]).to eq(invoice_item.item_id)
+    expect(found_invoice_item["invoice_id"]).to eq(invoice_item.invoice_id)
+    expect(found_invoice_item["quantity"]).to eq(invoice_item.quantity)
+    expect(found_invoice_item["unit_price"]).to eq(invoice_item.unit_price)
+    expect(Time.zone.parse(found_invoice_item["created_at"]).to_s).to eq(invoice_item.created_at.to_s)
+    expect(Time.zone.parse(found_invoice_item["updated_at"]).to_s).to eq(invoice_item.updated_at.to_s)
+  end
+
+  it "can return multiple records with matching id" do
+    create_list(:invoice_item, 2)
+
+    get "/api/v1/invoice_items/find_all?id=#{InvoiceItem.first.id}"
+
+    found_invoice_items = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(found_invoice_items).to be_a(Array)
+    expect(found_invoice_items.count).to eq(1)
```

Comment:
In your other `find_all` tests, you check the attribute that you searched for to confirm that you found the right record. It would be easy to do a quick check of the `id`s here to match the others